packages/builders/src/components/button/CustomIdButton.ts

```diff
@@ -0,0 +1,43 @@
+import { ComponentType, type APIButtonComponentWithCustomId } from 'discord-api-types/v10';
+import { Mixin } from 'ts-mixer';
+import { ButtonBuilder } from './Button.js';
+import { URLOrCustomIdButtonMixin } from './mixins/URLOrCustomIdButtonMixin.js';
+
+export type CustomIdButtonStyle = APIButtonComponentWithCustomId['style'];
+
+/**
+ * A builder that creates API-compatible JSON data for buttons with custom IDs.
+ */
+export class CustomIdButtonBuilder extends Mixin(
+	ButtonBuilder<APIButtonComponentWithCustomId>,
+	URLOrCustomIdButtonMixin,
+) {
+	protected override readonly data: Partial<APIButtonComponentWithCustomId>;
+
+	public constructor(data: Partial<APIButtonComponentWithCustomId> = {}) {
+		super();
+		this.data = { ...structuredClone(data), type: ComponentType.Button };
+	}
+
+	/**
+	 * Sets the style of this button.
+	 *
+	 * @param style - The style to use
+	 */
+	public setStyle(style: CustomIdButtonStyle) {
```

Comment:
This just feels like the `ChatInputCommandBuilder` vs `ContextMenuCommandBuilder.setType()` thing all over again.

I think we should have separate classes per button style, i.e. `PrimaryButton`, `SecondaryButton`, etc.

Personally it feels weird that to make a `Premium` button you do `new SKUIdButton()`, but for a primary one you'd have to use `new CustomIdButton().setStyle(ButtonStyle.Primary)`